tox.ini

```diff
@@ -0,0 +1,22 @@
+[tox]
+envlist = py27-django{18,111}
+
+[testenv]
+setenv =
+    PYTHONHASHSEED = 0
+passenv =
+    EDX_PLATFORM_SETTINGS
+    EDXAPP_TEST_MONGO_HOST
+deps =
+    django18: Django>=1.8,<1.9
+    django111: Django>=1.11,<2
```

Comment:
I'm not sure if it's relevant right now, but if anything we have in requirements has a different Django version specified in setup.py's install_requires it could overwrite this version. See: https://github.com/tox-dev/tox/issues/42

It might be prudent to put these lines down into explicit pip install commands to make absolutely sure the correct version of Django is installed last.

Reply:
So far, just using tox's injection of the Django version has worked, i.e. installed the correct version. Maybe that's because we've already converted all (known) dependencies that have specified a Django version. But I would expect pip to just fail on incompatible versions instead of actually switching the top-level explicitly-specified Django version.

If you feel we need it, how would I specify different versions based on the testenv? Can you split out commands for different testenvs?

Reply:
If it's working I suppose we don't need it. Where I've run into issues is in xblock repos where things were installed across different runs up pip. If you do run into weird issues where Django ends up being the wrong version you can run different commands based on env, just like in `deps`. See [xblock-utils](https://github.com/edx/xblock-utils/blob/master/tox.ini) for a particularly gnarly example.

Reply:
I see - xblock-utils doesn't install Django at all until the testenv:commands section - gnarly!